src/controllers/Realtime.ts

```diff
@@ -0,0 +1,259 @@
+    index: string,
+    collection: string,
+    message: JSONObject,
+    options: { queuable?: boolean, _id?: string } = {}
+  ): Promise<boolean> {
+    const request = {
+      index,
+      collection,
+      body: message,
+      action: 'publish',
+      _id: options._id
+    };
+
+    return this.query(request, options)
+      .then(response => response.result.published);
+  }
+
+  /**
+   * Subscribes by providing a set of filters: messages, document changes
+   * and, optionally, user events matching the provided filters will generate
+   * real-time notifications.
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/realtime/subscribe/
+   * @see https://docs.kuzzle.io/sdk/js/7/essentials/realtime-notifications/
+   *
+   * @param index Index name
+   * @param collection Collection name
+   * @param filters Optional subscription filters (@see https://docs.kuzzle.io/core/2/guides/cookbooks/realtime-api)
+   * @param callback Callback function to handle notifications
+   *
+   * @returns A string containing the room ID
+   */
+  subscribe (
+    index: string,
+    collection: string,
+    filters: JSONObject,
+    callback: (Notification) => void | Promise<void>,
+    options: {
+      /**
+       * Subscribe to document entering or leaving the scope. (default: 'all')
+       */
+      scope?: ScopeOption;
+      /**
+       * Subscribe to users entering or leaving the room. (default: 'none')
+       */
+      users?: UserOption;
+      /**
+       * Subscribe to notifications fired by our own queries. (default: true)
+       */
+      subscribeToSelf?: boolean;
+      /**
+       * Subscription information sent alongside notifications
+       */
+      volatile?: JSONObject;
+    } = {}
+  ): Promise<string> {
+    const room = new Room(this, index, collection, filters, callback, options);
+
+    return room.subscribe()
+      .then(() => {
+        if (!this._subscriptions.has(room.id)) {
+          this._subscriptions.set(room.id, []);
+        }
+        this._subscriptions.get(room.id).push(room);
+        return room.id;
+      });
+  }
+
+  /**
+   * Removes a subscription
+   *
+   * @param roomId Subscription room ID
+   * @param options Additional options
+   *    - `queuable` If true, queues the request during downtime, until connected to Kuzzle again
+   */
+  unsubscribe (
+    roomId: string,
+    options: { queuable?: boolean } = {}
+  ): Promise<void> {
+    const request = {
+      action: 'unsubscribe',
+      body: { roomId }
+    };
+
+    return this.query(request, options)
+      .then(response => {
+        const rooms = this._subscriptions.get(roomId);
+
+        if (rooms) {
+          for (const room of rooms) {
+            room.removeListeners();
+          }
+
+          this._subscriptions.delete(roomId);
+        }
+
+        return response.result;
+      });
+  }
+
+  /**
+   * Internal method.
+   *
+   * Should be called on network error or disconnection
+   */
+  disconnected () {
```

Comment:
🤔 can't we make this private/protected, since it's an internal method? Oh, it probably needs to be called by the Protocol class so that might be a reason to leave it public.
Ok, so we have something similar to the "loose-coupling" problem we had in the core.
By the way, the naming of this method looks confusing to me. If it was a handler, I'd call it `onDisconnected()`, which tells that "what goes in this block gets executed when the connection is closed", which is OK, but I'd prefer the name of the method tells _what the method does_, i.e. "delete all the subscriptions of this instance of the SDK".
I posted [a similar comment on a PR in the core](https://github.com/kuzzleio/kuzzle/pull/1735#issuecomment-671427457), so I'd like to leave my 2 cents on the naming of this method (if it's not a breaking change)

* deleteAllSubscriptions()
* deleteAllRooms()

👋

Reply:
I take advantage of existing events and I renamed the methods to `saveSubscriptions` and `resubscribe`